### PR TITLE
test(crypto): add test for status of connection Established

### PR DESCRIPTION
**Pull request by NamsooCho**

### CODE REVIEW

File: src/toxcore/net_crypto/mod.rs [hunk 1]
```diff
@@ -2488,4 +2488,41 @@ mod tests {
...
+        let session_precomputed_key = precompute(&peer_session_pk, &session_sk);
+        connection.status = ConnectionStatus::Established {
+            sent_nonce: gen_nonce(),
+            received_nonce,
+            peer_session_pk,
+            session_precomputed_key: session_precomputed_key,
+        };
+
+        // send status packet with connection.status is Established
+        assert!(net_crypto.send_status_packet(&mut connection).wait().is_ok());
```
Comment (kurnevsky) [inline src/toxcore/net_crypto/mod.rs]: Check that nothing was sent, like: https://github.com/tox-rs/tox/blob/d3d0aa241e1779d15f5ea1655824efd459ceed3a/src/toxcore/dht/server/mod.rs#L1864-L1867

Comment (NamsooCho) [reply src/toxcore/net_crypto/mod.rs]: done

File: src/toxcore/net_crypto/mod.rs [hunk 2]
```diff
@@ -2488,4 +2488,46 @@ mod tests {
...
+
+        // Necessary to drop udp_tx so that udp_rx.collect() can be finished
+        drop(net_crypto);
```
Comment (kpp) [inline src/toxcore/net_crypto/mod.rs]: udp_tx or net_crypto?

Comment (NamsooCho) [reply src/toxcore/net_crypto/mod.rs]: drop(net_crypto.udp_tx) is correct.

**Pull request merged**